ui/app/controllers/settings/tokens.js

```diff
@@ -1,20 +1,52 @@
 import Ember from 'ember';
 
-const { Controller, inject } = Ember;
+const { Controller, inject, computed, getOwner } = Ember;
 
 export default Controller.extend({
   token: inject.service(),
 
-  actions: {
-    setTokenProperty(property, event) {
-      this.get('token').set(property, event.currentTarget.value);
-    },
+  tokenRecord: null,
+  secret: computed.reads('token.secret'),
+
+  tokenIsValid: false,
+  tokenIsInvalid: false,
 
+  actions: {
     clearTokenProperties() {
       this.get('token').setProperties({
         secret: undefined,
         accessor: undefined,
       });
+      this.setProperties({
+        tokenIsValid: false,
+        tokenIsInvalid: false,
+        tokenRecord: null,
+      });
+    },
+
+    verifyToken() {
+      const { secret } = this.getProperties('secret', 'accessor');
+      const TokenAdapter = getOwner(this).lookup('adapter:token');
+
+      this.set('token.secret', secret);
+
+      TokenAdapter.findSelf().then(
+        token => {
+          this.setProperties({
+            tokenIsValid: true,
+            tokenIsInvalid: false,
+            tokenRecord: token,
+          });
+        },
+        () => {
+          this.set('token.secret', null);
+          this.setProperties({
+            tokenIsInvalid: true,
```

Comment:
ha swapping the order here took me a second (before this it's always valid, invalid, record) - maybe swap it here for consistency?

Reply:
👍 I dig it.